cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataRecord.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.metadata;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents the complete metadata of a {@link MetadataEntity} including its properties, tags for a given
+ * {@link MetadataScope}
+ */
+public class MetadataRecord {
+  private final MetadataEntity metadataEntity;
```

Comment:
Should the entity be part of the record? If it does, that means you can't compare the metadata of two different entities directly.